_templates/l10n.php

```diff
@@ -35,334 +37,368 @@ function list_langs() {
         'zh_CN' => '简体中文',
         'zh_TW' => '繁體中文',
     );
-}
 
-function lang_dir($lang) {
-    return dirname(__FILE__).'/../lang/'.$lang;
-}
+    protected $lang = 'en';
 
-function is_lang($lang) {
-    if (!is_string($lang)) {
-        return false;
-    }
-    if (!preg_match('#^[a-z]{2}(_([A-Z]{2}|[A-Z][a-z]+))?$#', $lang)) {
-        return false;
+    protected $domain = null;
+
+    public function __construct($lang = null) {
+        if (empty($lang)) {
+            $lang = $this->get_page_lang();
+        }
+        $this->lang = $lang;
     }
-    if ($lang == 'en') {
-        return true;
+
+    public function init() {
+        global $sitewide, $page; // Global site variables
```

Comment:
Is decided to keep these global variables because they are global site variables, ie. _safe_ (used on every page, breaks the whole website if corrupted). Do you think it's still important to pass them as arguments?